students/17th/team1/jaehyeonpark/user/views.py

```diff
@@ -0,0 +1,97 @@
+                return JsonResponse({'message':'NO_VALUE_ERROR'}, status=400)
+            
+            email_regex = re.compile("^.+@+.+\.+.+$")
+            
+            if not email_regex.match(email):
+                return JsonResponse({'message':'EMAIL_VALIDATION_ERROR'}, status=400)
+            
+            if User.objects.filter(email=email).exists():
+                return JsonResponse({'message':'EMAIL_EXISTS'}, status=400)
+
+            if User.objects.filter(phone_number=phone_number).exists():
+                return JsonResponse({'message':'PHONE_NUMBER_EXISTS'}, status=400)
+
+            if User.objects.filter(account=account).exists():
+                return JsonResponse({'message':'ACCOUNT_EXISTS'}, status=400)
+            
+            if len(password) < PASSWORD_MINIMUM_LENGTH:
+                return JsonResponse({'message':'PASSWORD_VALIDATION_ERROR'}, status=400)
+     
+            hashed_password = bcrypt.hashpw(password.encode('utf-8'), bcrypt.gensalt())
+            encrypted_password = hashed_password.decode('utf-8')
+            
+            User.objects.create(email=email, password=encrypted_password, phone_number=phone_number, account=account)
+
+        except json.decoder.JSONDecodeError:
+            return JsonResponse({'message':'JSON_DECODE_ERROR'}, status=400)
+        
+        except KeyError:
+            return JsonResponse({'message':'KEY_ERROR'}, status=400)
+        
+        except DataError:
+            return JsonResponse({'message':'DATA_ERROR'}, status=400)
+
+        except IntegrityError:
+            return JsonResponse({'message':'INTEGRITY_ERROR'}, status=400)
+        
+        return JsonResponse({'message':'SUCCESS'}, status=200)
+
+class SignInView(View):
+    def post(self, request):
+        try:
+            data         = json.loads(request.body)
+            email        = data.get('email')
+            phone_number = data.get('phone_number')
+            account      = data.get('account')
+            password     = data['password']
+            user_account = [email, phone_number, account]
+            
+            valid_user_account_number = len(list(filter(lambda x:x, user_account)))
+
+            if valid_user_account_number > 1:
+                return JsonResponse({'message':'TOO_MANY_USER_INFO'}, status=400)
+
+            print(email, account, phone_number)
+
+            if not (email or account or phone_number):
+                return JsonResponse({'message':'KEY_ERROR'}, status=400)
+            
+            if User.objects.filter(Q(email=email)|Q(phone_number=phone_number)|Q(account=account)).exists():
+                user = User.objects.get(Q(email=email)|Q(phone_number=phone_number)|Q(account=account))
+
+                if bcrypt.checkpw(password.encode('utf-8'), user.password.encode('utf-8')):
+                    user_id = user.id
+                    access_token = jwt.encode({'user_id': user_id}, SECRET, algorithm='HS256')
```

Comment:
```suggestion
                    access_token = jwt.encode({'user_id': user.id}, SECRET, algorithm='HS256')
```
재사용성 없는 변수 사용은 지양합시다~!

Reply:
네. 의미없는 변수 선언으로 메모리 낭비하는 부분을 고치겠습니다.